src/api/lib/backend/api.rb

```diff
@@ -27,5 +27,19 @@ def self.meta_from_deleted_project(project, revision)
     def self.trigger_services(project, package, user)
       Backend::Connection.post("/source/#{CGI.escape(project)}/#{CGI.escape(package)}?cmd=runservice&user=#{CGI.escape(user)}")
     end
+
+    # Copy a package into another project
+    def self.copy_package(target_project, target_package, source_project, source_package, login, options = {})
+      path = "/source/#{CGI.escape(target_project)}/#{CGI.escape(target_package)}"
+      query_hash = { cmd: :copy, oproject: source_project, opackage: source_package, user: login }
+      query_hash.merge!(options.slice(:keeplink, :expand, :comment))
+      path += "?#{query_hash.to_query}"
+      Backend::Connection.post(path)
+    end
+
+    # Writes the link information of a package
+    def self.write_link_of_package(project, package, login, xml)
+      Backend::Connection.put("/source/#{CGI.escape(project)}/#{CGI.escape(package)}/_link?user=#{CGI.escape(login)}", xml)
```

Comment:
This hast to be URI.escape for the first project and package or it breaks on some chars.

Reply:
Which chars?
URI.escape is going to be deprecated (see https://github.com/alexreisner/geocoder/issues/609)